src/operator/main.go

```diff
@@ -202,77 +202,139 @@ func main() {
+		clusterName := viper.GetString(operatorconfig.AWSRolesAnywhereClusterName)
 
-		azureAgent, err := azureagent.NewAzureAgent(signalHandlerCtx, config)
-		if err != nil {
-			logrus.WithError(err).Panic("failed to initialize Azure agent")
+		awsOptions = append(awsOptions, awsagent.WithRolesAnywhere(trustAnchorArn, trustDomain, clusterName))
+
+		if viper.GetBool(operatorconfig.AWSUseSoftDeleteStrategyKey) {
+			awsOptions = append(awsOptions, awsagent.WithSoftDeleteStrategy())
 		}
-		iamAgents = append(iamAgents, azureAgent)
+	}
+	awsAgent, err := awsagent.NewAWSAgent(ctx, awsOptions...)
+	if err != nil {
+		logrus.WithError(err).Panic("failed to initialize AWS agent")
+	}
+
+	return awscredentialsagent.NewAWSCredentialsAgent(awsAgent,
+		viper.GetBool(operatorconfig.AWSUseSoftDeleteStrategyKey),
+		viper.GetBool(operatorconfig.EnableAWSRolesAnywhereKey),
+		viper.GetString(operatorconfig.AWSRolesAnywhereTrustAnchorARNKey))
+}
+
+func initGCPCredentialsAgent(ctx context.Context, client controllerruntimeclient.Client) *gcpcredentialsagent.Agent {
+	gcpAgent, err := gcpagent.NewGCPAgent(ctx, client)
+	if err != nil {
+		logrus.WithError(err).Panic("failed to initialize GCP agent")
 	}
 
-	azureGCPserviceAccountReconciler := serviceaccounts.NewServiceAccountReconciler(client, iamAgents)
-	if err = azureGCPserviceAccountReconciler.SetupWithManager(mgr); err != nil {
-		logrus.WithField("controller", "ServiceAccount").WithError(err).Panic("unable to create controller")
+	return gcpcredentialsagent.NewGCPCredentialsAgent(gcpAgent)
+}
+
+func initAzureCredentialsAgent(ctx context.Context) *azurecredentialsagent.Agent {
+	config := azureagent.Config{
+		SubscriptionID: viper.GetString(operatorconfig.AzureSubscriptionIdKey),
+		ResourceGroup:  viper.GetString(operatorconfig.AzureResourceGroupKey),
+		AKSClusterName: viper.GetString(operatorconfig.AzureAksClusterNameKey),
+	}
+
+	azureAgent, err := azureagent.NewAzureAgent(ctx, config)
+	if err != nil {
+		logrus.WithError(err).Panic("failed to initialize Azure agent")
 	}
+	return azurecredentialsagent.NewAzureCredentialsAgent(azureAgent)
+}
 
-	if viper.GetBool(operatorconfig.EnableAzureServiceAccountManagementKey) ||
-		viper.GetBool(operatorconfig.EnableGCPServiceAccountManagementKey) ||
-		viper.GetBool(operatorconfig.EnableAWSServiceAccountManagementKey) {
-		podReconciler := pods.NewPodReconciler(client)
-		if err = podReconciler.SetupWithManager(mgr); err != nil {
+func setupIAMAgents(ctx context.Context, mgr ctrl.Manager, client controllerruntimeclient.Client, podNamespace string) {
+	awsIAMEnabled := viper.GetBool(operatorconfig.EnableAWSServiceAccountManagementKey)
+	azureIAMEnabled := viper.GetBool(operatorconfig.EnableAzureServiceAccountManagementKey)
+	gcpIAMEnabled := viper.GetBool(operatorconfig.EnableGCPServiceAccountManagementKey)
+
+	if !awsIAMEnabled && !azureIAMEnabled && !gcpIAMEnabled {
+		return
+	}
+
+	var awsCredentialsAgent *awscredentialsagent.Agent
```

Comment:
Can we move the webhook setup into the awsIAMEnabled (and others) block?
So that:
1. We don't have a scope where the agents are uninitialized
2. All the agent initialization code is written close together